Update variable structure and include testinfra

jpward1981 opened the pull request:
Included the testing for debian based nodes
Update testinfra to confirm that test packages are installed

Commit:
Update the packages repo to install packages based on a dictionary passed to the role during run time.

ragingpastry commented on tasks/redhat.yml:
```diff
@@ -1,4 +1,12 @@
+    state: latest
+  when:
+    - packages.update_all is defined
+    - packages.update_all == True
```
Lint error here. ansible-lint wants this as "true" instead of "True"

We could also just cast packages.update_all to a bool and then put the check like this:

`packages.update_all | bool` or `packages.update_all` (without the check == True)

Either way we should modify this so that it will pass the lint stage

jpward1981 replied:
I have corrected this in the last commit

ragingpastry commented on tasks/debian.yml:
```diff
@@ -1,4 +1,11 @@
 - name: Install deb packages
   apt:
-    name: "{{ packages }}"
+    update_cache: "{{ packages.debian.update_cache | default('yes') }}"
```
This will break when run against Debian instances that don't have a defined debian dictionary under packages. I could see this coming up when this role is ran against a diverse inventory that only takes action against certain OS targets. For example, if we have Debian and RHEL hosts in our inventory but we are only installing a RHEL package that is included by default on Debian.

This won't be a problem in Ansible 2.8 because they will start including support for defining default values for nested dicts (https://docs.ansible.com/ansible/devel/porting_guides/porting_guide_2.8.html#jinja-undefined-values), however until then we will have to use a different syntax.

jpward1981 replied:
Updated the when logic to handle the missing variables. I also updated the testing to test these kinds of scenarios.

Commit:
Correct some issues in the logs and add more tests

Add some additional  test scenarios on converge tests
Add additional test for test infra to confirm package installation
and for EL test yum cache creating times
Inproved error handling when something is not defined in the
packages dictonary.
Added Travis CI testing
